lib/commands/server_handshake.js

```diff
@@ -69,38 +70,69 @@ ServerHandshake.prototype.readClientReply = function(packet, connection) {
 
 ServerHandshake.prototype.dispatchCommands = function(packet, connection) {
   // command from client to server
+  var knownCommand = true;
   var commandCode = packet.readInt8();
   switch (commandCode) {
   case CommandCode.QUIT:
-    connection.emit('quit');
-    // connection.stream.end();
+    if (connection.listeners('quit').length) {
+      connection.emit('quit');
+    } else {
+      connection.stream.end();
```

Comment:
should be ok. This just ends underlying stream ( tcp/unix socket etc). I think this is what mysql server does, no extra data sent in response